### DISCUSSION TITLE
How to launch a websocket server?

### DISCUSSION THREAD

**Discussion by spiffytech** [solidjs/solid-start#637] [Q&A]
I'd like to launch a websocket server on my backend to handle real-time features for my app.

I also need to intercept the HTTP request that initiates a websocket upgrade so I can process the user's auth cookie before allowing the connection upgrade.

How can I do this in SolidStart?

**Comment by marbemac** [comment]
There is a `with-websocket` example - perhaps start there to see if it helps to get you going? 

https://github.com/solidjs/solid-start/tree/main/examples/with-websocket

**Reply by spiffytech** [reply]
I've tried to piece together what's happening in that project, but I can't get it to work. If I `npm init solid` using the `with-websocket` template, DevTools shows the websocket request stays `pending` and never sends/receives messages. If I try reimplementing the important bits in a clean project, I see the same behavior.

**Reply by floralpetals** [reply]
I got the same issue, even when I clone the example project.
